src/core/Elsa.Core/Services/WorkflowHost.cs

```diff
@@ -75,8 +76,12 @@ public class WorkflowHost : IWorkflowHost
         public async Task<WorkflowExecutionContext> RunWorkflowDefinitionAsync(string workflowDefinitionId, string? activityId, object? input = default, string? correlationId = default, CancellationToken cancellationToken = default)
         {
             var workflow = await workflowRegistry.GetWorkflowAsync(workflowDefinitionId, VersionOptions.Published, cancellationToken);
+            if (workflow == null)
+            {
+                logger.LogError("Workflow definition {WorkflowInstanceId} either not registered or not published.", workflowDefinitionId);
+                throw new WorkflowException("The triggered workflow definition either not registered or not published.");
```

Comment:
Please rephrase as "The specified workflow definition is either not registered or not published.". Thanks!

Reply:
@sfmskywalker I have rephrase the error message.

Reply:
Thanks!